docs/topics/using-builders-with-builder-inference.md

```diff
@@ -0,0 +1,253 @@
+[//]: # (title: Using builders with builder type inference)
+
+Kotlin supports _builder type inference_ (or builder inference), which is a type inference flavour that comes in handy 
+when writing generic builders. It helps the compiler infer the type arguments of a builder call based on the type information 
```

Comment:
> when writing generic builders

Actually, it comes in handy when _using_ generic builders. BI works at call site.

Reply:
Maybe it's possible to use something common like "when working with generic builders"

Reply:
Yeap, changed to "when working with generic builders"